python/erdos/operator.py

```diff
@@ -6,85 +6,10 @@
 MAX_NUM_RUNTIME_SAMPLES = 1000
 
 
-class Operator(object):
-    """An :py:class:`Operator` is an abstract base class that needs to be
-    inherited by user-defined operators in order to be run in an ERDOS dataflow
-    graph.
-
-    A user-defined operator needs to inherit from :py:class:`Operator` and
-    implement :py:func:`Operator.__init__` and :py:func:`Operator.connect` in
-    order to be connected to the dataflow graph. For example, a `MapOperator`
-    that takes in a single input stream and outputs data on a single output
-    stream can be implemented as follows::
-
-        class MapOperator(erdos.Operator):
-            def __init__(self, read_stream, write_stream):
-                # Register callbacks on read streams and save write streams.
-                pass
-
-            @staticmethod
-            def connect(read_stream):
-                return erdos.WriteStream()
-
-    Instead of ERDOS invoking callbacks registered on the read streams, an
-    operator can also take control of the execution by overriding the
-    :py:func:`Operator.run` method as follows::
-
-        class MapOperator(erdos.Operator):
-            def run(self):
-                message = self.read_stream.read()
-                # Work on the message.
+class BaseOperator(object):
```

Comment:
Can the classes in this module inherit from [abstract base class](https://docs.python.org/3/library/abc.html)?